### ISSUE TITLE
Custom Email Template for Notifications being Overwritten

### ISSUE THREAD

**Issue by tplassman** [barrelstrength/craft-sprout-email#158]
### Description

If a custom template folder is defined for an email notification, the custom template is not loaded as the value for the field in subsequent loads of the notification edit screen.  This means that any subsequent changes to a notification will revert the custom template location to the default email template.

https://github.com/barrelstrength/craft-sprout-base-email/blob/v1/src/controllers/NotificationsController.php#L210

Line 210 will always override the emailTemplateId property even if it was previously set to a custom location.

### Steps to reproduce

1. Create custom notification entry.
2. Edit custom notification entry to use a custom email template folder.
3. Save the notification entry and reload the edit entry screen.
4. The custom email template will display the default email template instead of the custom folder.

### Additional info

- Craft version: 3.4.17.1
- PHP version: PHP 7.2
- Database driver & version: MySQL
- Plugins & versions: Sprout Forms v3.11.2

**Comment by BenParizek** [comment]
The ticket was filed in the right repo but for current workflow considerations, I've moved this to the sprout-email repo.

**Comment by tplassman** [comment]
Thanks @BenParizek.  My apologies for the duplicate issue.  It looks like similar issues have already been filed in this repo that I did not see.

Please let me know if there is anything I can help w/.

**Comment by BenParizek** [comment]
@tplassman This issue should be resolved in v4.4.2, however:

I'd also recommend you review all your other email settings. There was a bug in a migration that could reset all shared Sprout settings. The only way to fix this is to roll-back and re-run the updates or retrieve your old settings from a backup database.

If you need to review the shared Sprout settings in the db directly, we are currently using a different method to save settings that Craft does with the Project Config. The Sprout Settings would be in the `sprout_settings` table and the below SQL statement reflects what they should roughly look like for Sprout Email:

```
INSERT INTO `craft_sprout_settings` (`id`, `model`, `settings`, `dateCreated`, `dateUpdated`, `uid`) VALUES
('3', 'barrelstrength\\sproutbasesentemail\\models\\Settings', '{\"pluginNameOverride\":\"\",\"enableSentEmails\":\"1\",\"sentEmailsLimit\":\"5000\",\"cleanupProbability\":\"1000\"}', '2020-05-16 15:29:27', '2020-05-16 16:16:42', '37ef68ad-9f77-41dd-a3ce-013bec6b1b65'),
('4', 'barrelstrength\\sproutbaseemail\\models\\Settings', '{\"pluginNameOverride\":\"\",\"enableNotificationEmails\":\"1\",\"emailTemplateId\":\"\",\"enablePerEmailEmailTemplateIdOverride\":\"\"}', '2020-05-16 15:29:27', '2020-05-16 16:06:09', 'eac7828c-c42a-4f7a-b705-10e2d3329103');
```

_Note: For reasons I won't go into here, we currently manage some settings in this non-standard way (as the default way doesn't solve all the needs for our particular plugin use-case with some shared features between plugins). I recognize that this abstraction has caused more problems than are ideal and we're currently exploring ways to update the way settings are handled in our plugins to figure out how we can solve these issue and get things working in a more Craft-native way._

**Comment by tplassman** [comment]
@BenParizek,

Thanks for the quick response and action on this issue.  I will pull the update for our client today.

Also appreciate the heads up on the settings.  I will back up the existing settings in case of trouble during the update.

Thanks again!
Trevor
